### ISSUE TITLE
support for numpy v1.24

### ISSUE THREAD

**Issue by kbroch-rivosinc** [mhostetter/galois#471]

**Comment by mhostetter** [comment]
I merged a PR that I thought fixed this. However, Numba is tricky and uninstall NumPy 1.24.1 and uses 1.23.5 instead. I'm going to force push and remove those commits from `release/0.3.x`.

This will need to wait for Numba support.

```
matt@DESKTOP-I79NAGP:/mnt/c/Users/matth/repos/galois$ python3 -m pip install -U numpy
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: numpy in /home/matt/.local/lib/python3.8/site-packages (1.23.5)
Collecting numpy
  Using cached numpy-1.24.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
Installing collected packages: numpy
  Attempting uninstall: numpy
    Found existing installation: numpy 1.23.5
    Uninstalling numpy-1.23.5:
      Successfully uninstalled numpy-1.23.5
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
numba 0.56.4 requires numpy<1.24,>=1.18, but you have numpy 1.24.1 which is incompatible.
galois 0.2.1.dev2+g13770e0d.d20221208 requires numpy<1.24,>=1.21.0, but you have numpy 1.24.1 which is incompatible.
Successfully installed numpy-1.24.1
matt@DESKTOP-I79NAGP:/mnt/c/Users/matth/repos/galois$ python3 -m pip install -U numba
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: numba in /home/matt/.local/lib/python3.8/site-packages (0.56.4)
Requirement already satisfied: importlib-metadata in /home/matt/.local/lib/python3.8/site-packages (from numba) (4.11.1)
Requirement already satisfied: setuptools in /home/matt/.local/lib/python3.8/site-packages (from numba) (65.1.0)
Requirement already satisfied: llvmlite<0.40,>=0.39.0dev0 in /home/matt/.local/lib/python3.8/site-packages (from numba) (0.39.0)
Collecting numpy<1.24,>=1.18
  Using cached numpy-1.23.5-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.1 MB)
Requirement already satisfied: zipp>=0.5 in /home/matt/.local/lib/python3.8/site-packages (from importlib-metadata->numba) (3.8.1)
Installing collected packages: numpy
  Attempting uninstall: numpy
    Found existing installation: numpy 1.24.1
    Uninstalling numpy-1.24.1:
      Successfully uninstalled numpy-1.24.1
Successfully installed numpy-1.23.5
matt@DESKTOP-I79NAGP:/mnt/c/Users/matth/repos/galois$ python3
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
matt@DESKTOP-I79NAGP:/mnt/c/Users/matth/repos/galois$ 
matt@DESKTOP-I79NAGP:/mnt/c/Users/matth/repos/galois$ ipython3 
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.1.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import numba; numba.__version__
Out[1]: '0.56.4'

In [2]: import numpy; numpy.__version__
Out[2]: '1.23.5'
```

**Comment by iyanmv** [comment]
I've tested [this PR](https://github.com/numba/numba/pull/8691) that got merged today in numba, and all galois 0.3.3' tests work fine with numpy 1.24. So probably only a few more days until numba has a new release version compatible with the latest numpy 🎉

**Comment by mhostetter** [comment]
Great! We will update `galois` and release as 0.3.4 whenever Numba 0.57 is available through `pip`. Then we will have support for NumPy 1.24 and Python 3.11.
